lib/locks/bakery/bakery_lock_normal.c

```diff
@@ -222,6 +222,7 @@ void bakery_lock_release(bakery_lock_t *lock)
 	unsigned int is_cached = read_sctlr_el3() & SCTLR_C_BIT;
 
 	my_bakery_info = get_bakery_info(plat_my_core_pos(), lock);
+	read_cache_op(my_bakery_info, is_cached);
```

Comment:
This `read_cache_op` is only required when the assert is compiled in. So probably you can define a function to invoke within assert like:
```
static inline int is_lock_acquired(const bakery_info_t *my_bakery_info, int is_cached)
{
    read_cache_op(my_bakery_info, is_cached);
    return !!(bakery_ticket_number(my_bakery_info->lock_data));
}
```
And invoke this function within assert at line 226 and 112.